### PR TITLE
Created Post date component.

### PR DESCRIPTION
**Pull request by Danishkhurshid**

### CODE REVIEW

**Comment by Danishkhurshid** [comment]
@bmarshall511 Created post-date component. Abstracted the reusable code from Text block and put it in shared folder. Kindly review.

File: build/block-library/post-date/block.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      "type": "object"
+    }
+  },
+  "editorScript": "file:./index.js",
+  "editorStyle": "file:./index.css",
+  "style": "file:./style-index.css",
+  "supports": {
+    "customClassName": false
+  }
+}
```
Comment (Danishkhurshid) [inline build/block-library/post-date/block.json:35]: @bmarshall511 there is an EOD missing here. But for some odd reason the build keeps generating file with no last line even though i have tried to add few empty lines.

File: build/block-library/post-date/block.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "https://json.schemastore.org/block.json",
+  "apiVersion": 2,
+  "name": "gutenberg-ant-design/post-date",
+  "version": "0.1.0",
+  "title": "Post Date",
+  "description": "Add the date of this post inside a Query loop.",
```
Comment (bmarshall511) [inline build/block-library/post-date/block.json:7]: @Danishkhurshid This isn't & shouldn't be specific to a query loop.

File: build/block-library/post-date/block.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (bmarshall511) [inline build/block-library/post-date/block.json:1]: @Danishkhurshid Missing the icon property, please add one that makes sense.

File: src/block-library/post-date/edit.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+	// Used by the Gutenberg editor to save & output blocks properly
+	const blockProps = useBlockProps({
+		className: `gutenberg-ant-design--${clientId}`,
+	});
+
+	// Merge the default attributes with the saved ones
+	const savedAttributes = { ...textAttributes, ...attributes };
+
+	// Component states
+	const [popoverAnchor, setPopoverAnchor] = useState(null);
```
Comment (bmarshall511) [inline src/block-library/post-date/edit.js:55]: @Danishkhurshid What's this being used for?

**Review by bmarshall511 — CHANGES_REQUESTED** [review]

**Comment by Danishkhurshid** [comment]
@bmarshall511 Updated the PR. I had made this specific to the query loop. Added the functionality so it can be used outside of the query loop. Please review.

File: src/block-library/post-date/edit.js [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+  });
+
+  // Merge the default attributes with the saved ones
+  const savedAttributes = { ...textAttributes, ...attributes };
+
+  // Component states
+	
+  const isDescendentOfQueryLoop = Number.isFinite(queryId);
+  // used to display date popup that let's the user to edit the date outside Query loop
+  const [popoverAnchor, setPopoverAnchor] = useState(null);
```
Comment (bmarshall511) [inline src/block-library/post-date/edit.js:65]: @Danishkhurshid What pop-up is this referring to? Not seeing it when I play around with the component.

**Review by bmarshall511 — CHANGES_REQUESTED** [review]

**Comment by Danishkhurshid** [comment]
@bmarshall511 this displays when you click on the date and edit it. It's in the toolbar. 
ss

**Comment by bmarshall511** [comment]
cool, thanks @Danishkhurshid. looks good!

**Pull request merged**